### PR TITLE
SingleColumn: Fix vectorisation of nested else-if bodies

### PR DESCRIPTION
**Pull request by mlange05**
The recent special-casing for else-if branches in the SCC vectorisation only works for single-level nesting, but skips any further nested else-ifs and the final else-branch. TO fix this, i've introduced a `else_bodies` property on `Conditional` nodes that returns the nested bodies for multi-conditionals. Using this, this we can then apply vector-section extraction for every else body in a deeply nested multi-conditional.

### CODE REVIEW

File: loki/ir/nodes.py [hunk 1]
```diff
@@ -704,6 +704,16 @@ def __repr__(self):
...
         return 'Conditional::'
 
+    @property
+    def else_bodies(self):
+        """
+        Return all nested node tuples in the ``ELSEIF``/``ELSE`` part
+        of the conditional chain.
+        """
+        if self.has_elseif:
+            return (self.else_body[0].body,) + self.else_body[0].else_bodies
```
Comment (reuterbal) [inline loki/ir/nodes.py:714]: Now I have understood - very neat indeed 😄

File: loki/transformations/single_column/vector.py [hunk 2]
```diff
@@ -141,7 +141,11 @@ def extract_vector_sections(cls, section, horizontal):
                 # as 'Conditional's rely on the fact that the first element of the 'else_body'
                 # (if 'has_elseif') is a Conditional itself
                 if separator.has_elseif and separator.else_body:
```
Comment (reuterbal) [inline loki/transformations/single_column/vector.py:143]: Since `separator.else_bodies` should, even in the `has_elseif==False` case, return the else_body correctly, we should be able to remove this check here and always use the new implementation, no?

**Review by reuterbal — COMMENTED** [review]
Only a comment for now, but changes look very good. I think your solution actually allows to simplify the vector sections extraction again, removing the need to check for `has_elseif`.

I will let the nightly CI run against this branch to confirm whether it fixes the problem.

**Comment by mlange05** [comment]
Thanks for the early review; you were right, the logic could be drastically simplified. Unfortunately, last nights version was still buggy, but I've just rebase-pushed a clean version that should bring the tco399 ecphys regression back (confirmed locally). Please have another look, but should be ready now.

File: loki/transformations/single_column/vector.py [hunk 3]
```diff
@@ -140,12 +140,8 @@ def extract_vector_sections(cls, section, horizontal):
                 # we need to prevent that the whole 'else_body' is wrapped in a section,
                 # as 'Conditional's rely on the fact that the first element of the 'else_body'
                 # (if 'has_elseif') is a Conditional itself
```
Comment (MichaelSt98) [inline loki/transformations/single_column/vector.py:142]: Maybe rephrase this comment/explanation?

```suggestion
                # we need to prevent that all (possibly nested) 'else_bodies' are completely wrapped as a section,
                # as 'Conditional's rely on the fact that the first element of each 'else_body'
                # (if 'has_elseif') is a Conditional itself
```

**Review by MichaelSt98 — APPROVED** [review]
Very nice indeed!

**Review by reuterbal — APPROVED** [review]
The fixed fix seems to have resolved the problems in ifs-bundle. Many thanks for this!

I have accepted the comment suggestion from @MichaelSt98 as well.
